src/components/OrdersWidget/OrderRow.tsx

```diff
@@ -152,44 +91,156 @@ function calculatePrice(_numerator?: string | null, _denominator?: string | null
+      <div className="warning">{isOverBalance && <FontAwesomeIcon icon={faExclamationTriangle} />}</div>
+    </div>
+  )
+}
+
+const Expires: React.FC<Pick<Props, 'order' | 'pending'>> = ({ order, pending }) => {
   const { isNeverExpires, expiresOn } = useMemo(() => {
     const isNeverExpires = isBatchIdFarInTheFuture(order.validUntil)
     const expiresOn = isNeverExpires ? '' : formatDateFromBatchId(order.validUntil)
+
     return { isNeverExpires, expiresOn }
   }, [order.validUntil])
 
+  return <div>{isNeverExpires ? <Highlight color={pending ? 'grey' : ''}>Never</Highlight> : expiresOn}</div>
+}
+
+async function fetchToken(
+  tokenId: number,
+  orderId: string,
+  networkId: number,
+  setFn: React.Dispatch<React.SetStateAction<TokenDetails | null>>,
+): Promise<void> {
+  const token = await getTokenFromExchangeById({ tokenId, networkId })
+
+  // It is unlikely the token ID coming form the order won't exist
+  // Still, if that ever happens, store null and keep this order hidden
+  setFn(token)
+
+  // Also, inform the user this token failed and the order is hidden.
+  if (!token) {
+    toast.warn(
+      `Token id ${tokenId} used on orderId ${orderId} is not a valid ERC20 token. Order will not be displayed.`,
+    )
+  }
+}
+
+interface Props {
+  order: AuctionElement
+  isOverBalance: boolean
+  networkId: number
+  pending?: boolean
+}
+
+const OrderRow: React.FC<Props> = props => {
+  const { order, networkId, pending = false } = props
+
+  // Fetching buy and sell tokens
+  const [sellToken, setSellToken] = useSafeState<TokenDetails | null>(null)
+  const [buyToken, setBuyToken] = useSafeState<TokenDetails | null>(null)
+
+  useEffect(() => {
+    const errorMsg = 'Failed to fetch token'
+    fetchToken(order.buyTokenId, order.id, networkId, setBuyToken).catch(onErrorFactory(errorMsg))
+    fetchToken(order.sellTokenId, order.id, networkId, setSellToken).catch(onErrorFactory(errorMsg))
+  }, [networkId, order, setBuyToken, setSellToken])
+
   return (
-    <OrderRowWrapper className={'orderRow' + (pending ? ' pending' : '')}>
-      <PendingLink {...props} />
-      <div className="checked">
-        <input type="checkbox" />
-      </div>
-      <OrderDetails {...props} price={price} />
-      <UnfilledAmount {...props} unfilledAmount={unfilledAmount} unlimited={unlimitedAmount} />
-      <AccountBalance {...props} accountBalance={accountBalance} />
-      <Expires {...props} isNeverExpires={isNeverExpires} expiresOn={expiresOn} />
-    </OrderRowWrapper>
+    <>
+      {sellToken && buyToken && (
+        <OrderRowWrapper className={'orderRow' + (pending ? ' pending' : '')}>
+          <PendingLink {...props} />
+          <div className="checked">
+            <input type="checkbox" />
+          </div>
+          <OrderDetails {...props} sellToken={sellToken} buyToken={buyToken} />
+          <UnfilledAmount {...props} sellToken={sellToken} />
+          <AccountBalance {...props} sellToken={sellToken} />
```

Comment:
Are you sure about this kind of passing all the props wholesale to all the components? Usually you would pass props to one component or none at all. Otherwise many components will end up with props they don't need.
And if you know what props those components need, destructuring is better.

Reply:
Initially I was using this because I wasn't sure about the interface, in the first draft.
Then with `Pick<>`, the component only gets what it needs. Is it a big deal, performance wise?
I don't mind changing it, just looks cleaner this way.